src/_guides/language/concurrency/index.md

```diff
@@ -312,107 +311,144 @@ to implement isolates.
 }
+```
+
+The spawned isolate executes the following code:
 
-// Spawns an isolate and waits for the first message
-Future<Map<String, dynamic>> _parseInBackground() async {
-  final p = ReceivePort();
-  await Isolate.spawn(_readAndParseJson, p.sendPort);
-  return await p.first as Map<String, dynamic>;
+<?code-excerpt "lib/simple_worker_isolate.dart (spawned)"?>
+```dart
+Future<Map<String, dynamic>> _readAndParseJson() async {
+  final fileData = await File(filename).readAsString();
+  final jsonData = jsonDecode(fileData) as Map<String, dynamic>;
+  return jsonData;
 }
 ```
 
-The `_parseInBackground()` function contains the code that
-_spawns_ (creates and starts) the isolate for the background worker,
-and then returns the result:
+1. `Isolate.run()` spawns an isolate for the background worker,
+   while `main()` waits for the result.
 
-1. Before spawning the isolate, the code creates a `ReceivePort`,
-   which enables the worker isolate
-   to send messages to the main isolate.
+2. The new isolate executes the argument passed to `run()`:
+   the function `_readAndParseJson()`.
 
-2. Next is the call to `Isolate.spawn()`,
-   which creates and starts the isolate for the background worker.
-   The first argument to `Isolate.spawn()` is the function that
-   the worker isolate executes: `_readAndParseJson`.
-   The second argument is the `SendPort`
-   that the worker isolate can use to send messages to the main isolate.
-   The code doesn't _create_ a `SendPort`;
-   it uses the `sendPort` property of the `ReceivePort`.
+3. The `return` statement ends the computation,
+   shutting down the worker isolate and
+   sending `jsonData` back to the main isolate as the result.
 
-3. Once the isolate is spawned, the main isolate waits for the result.
-   Because the `ReceivePort` class implements `Stream`,
-   the [`first`][] property is an easy way to get
-   the single message that the worker isolate sends.
+4. The memory holding the result in the worker isolate is *transferred*
+   to the main isolate, *not copied*.
+   The worker isolate still performs a verification pass to ensure
+   the objects are allowed to be transferred.
 
-[`first`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-async/Stream/first.html
+`_readAndParseJson()` is an existing,
+asynchronous function that could just as easily
+run directly in the main isolate.
+But using `Isolate.run()` enables concurrency,
+so its computations are completely abstracted by the worker isolate
+and can complete without blocking the main isolate.
 
-The spawned isolate executes the following code:
+Because `Isolate.run()` allows code in the main isolate to continue running
+while its computations execute in a new one,
+its result is always returned as a `Future`.
+The original isolate doesn't care whether the computation
+running in the other isolate is synchronous or asynchronous,
+because it's being run concurrently either way.
```

Comment:
**suggestion:** This is a matter of tightening language again. Shorten the sentences, use active voice, avoid anthropomorphizing.
```suggestion
The `Isolate.run()` method returns a `Future`. 
This result returns because the main isolate continues to run
while computations execute in the new isolate.
The code running in the new isolate doesn't impact
 the original isolate.
```